keps/sig-auth/0034-20190123-harden-default-discovery-bindings.md

```diff
@@ -0,0 +1,71 @@
+---
+kep-number: 34
+title: Harden Default RBAC Discovery ClusterRole(Binding)s
+authors:
+  - "@dekkagaijin"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+  - sig-api-machinery
+reviewers:
+  - "@liggitt"
+  - "@tallclair"
+approvers:
+  - "@liggitt"
+  - "@tallclair"
+editor: TBD
+creation-date: 2019-01-23
+last-updated: 2019-01-23
+status: provisional
+---
+
+# Harden Default RBAC Discovery ClusterRole(Binding)s
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a KEP and for highlighting any additional information provided beyond the standard KEP template.
+[Tools for generating][] a table of contents from markdown are available.
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The aim of this change is to remove the `system:unauthenticated` subjects from the `system:discovery` and `system:basic-user` ClusterRoleBindings, while preserving unauthenticated access to a genuinely non-sensitive subset of the current APIs (e.g. `GET /healthz`, `GET /version`). This will improve the default privacy and security posture for new clusters.
+
+## Motivation
+
+One of the work items resulting from the [CVE-2018-1002105](https://github.com/kubernetes/kubernetes/issues/71411) [post-mortem](https://docs.google.com/document/d/1tGpqEM56cSBPfl-qxIs_CGdlABMZXEKEDhCI7eOwGT0/preview) was to [investigate hardening the default RBAC discovery ClusterRoleBindings](https://github.com/kubernetes/kubernetes/issues/72115) (i.e. `system:discovery` and `system:basic-user`) to limit potential avenues for similar attack. Additionally, the fact that API extensions are exposed by the default discovery bindings is surprising to some and represents a potential privacy concern (e.g. `/apis/self-flying-cars.lyft.io/v1/`).
```

Comment:
nit: The post mortem doesn't appear to be publicly available. If that's the case, I don't think it should be linked here.

Reply:
done